2006-07-26  Gwenole Beauchesne  <gbeauchesne@mandriva.com>

	* rt/Makefile (tests): Don't run tests in parallel on fine-grained
	SMT systems.

--- glibc-2.4.90/rt/Makefile.testsuite-rt-notparallel	2006-02-16 09:32:18.000000000 -0500
+++ glibc-2.4.90/rt/Makefile	2006-10-25 03:00:03.000000000 -0400
@@ -81,6 +81,16 @@ endif
 
 tst-mqueue7-ARGS = -- $(built-program-cmd)
 
+# XXX avoid timing issues on fine-grained SMT systems
+ifeq (powerpc, $(base-machine))
+no-parallel-testing = yes
+endif
+ifneq ($(filter %tests,$(MAKECMDGOALS)),)
+ifeq ($(no-parallel-testing),yes)
+.NOTPARALLEL:
+endif
+endif
+
 ifeq (yes,$(build-static-nss))
 otherlibs += $(nssobjdir)/libnss_files.a $(resolvobjdir)/libnss_dns.a \
              $(resolvobjdir)/libresolv.a
